### PR TITLE
Hotfix: Fix reschedule inside embed

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes "Reschedule Link"(/reschedule/{BOOKING_ID}) giving 404 with embed. This was reported on https://calendso.slack.com/archives/C01V37YT188/p1669806018267329. The alternative link of format /pro/30min?rescheduleUid={BOOKING_ID} was still working.
[Here is a loom showing that it's fixed and how it can be tested ](https://www.loom.com/share/619f7bf981fb43daa2c47fc8428de59a)

- I added a test in the embed test page as well(Added a seed for a hardcoded booking ID for the same).

**Environment**: Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Add a popup embed with calLink /reschedule/{BOOKING_ID} and test that the a reschedule is possible in the embed.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/lib/withEmbedSsr.tsx [hunk 1]
```diff
@@ -7,6 +7,28 @@ export type EmbedProps = {
 export default function withEmbedSsr(getServerSideProps: GetServerSideProps) {
   return async (context: GetServerSidePropsContext): Promise<GetServerSidePropsResult<EmbedProps>> => {
     const ssrResponse = await getServerSideProps(context);
+    const embed = context.query.embed;
+
+    if ("redirect" in ssrResponse) {
```
Comment (hariombalhara) [inline apps/web/lib/withEmbedSsr.tsx:12]: Reschedule flow currently redirects /reschedule/[uid] to /pro/30min?rescheduleUid=[uid].
So, let withEmbedSsr handle redirect property to ensure that embed context is properly forwareded.

File: apps/web/pages/_document.tsx [hunk 2]
```diff
@@ -3,10 +3,9 @@ import Document, { DocumentContext, Head, Html, Main, NextScript, DocumentProps
 type Props = Record<string, unknown> & DocumentProps;
 
 function toRunBeforeReactOnClient() {
-  const calEmbedMode =
-    location.search.includes("embed=") ||
-    /* Iframe Name */
-    window.name.includes("cal-embed");
+  const calEmbedMode = typeof new URL(document.URL).searchParams.get("embed") === "string";
```
Comment (hariombalhara) [inline apps/web/pages/_document.tsx:6]: Make embed detection even more reliable by  using searchParams that automatically handles all cases like
- a.com/?embed
-  a.com/?embed=
- a.com/?b=1&embed=
- a.com/?b=1&embed&c

File: apps/web/pages/reschedule/[uid].tsx [hunk 3]
```diff
@@ -42,13 +42,17 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
   if (!booking) {
     return {
       notFound: true,
+    } as {
```
Comment (hariombalhara) [inline apps/web/pages/reschedule/[uid].tsx:45]: Required to be used with `withEmbedSsr`. Without this `getServerSideProps` isn't passable to `withEmbedSsr`

File: packages/embeds/embed-core/index.html [hunk 4]
```diff
@@ -88,6 +88,7 @@ <h2>Other Popup Examples</h2>
       <button data-cal-namespace="popupTeamLinkLightTheme" data-cal-config='{"theme":"light"}' data-cal-link="team/seeded-team/collective-seeded-team-event">Book with Test Team[Light Theme]</button>
       <button data-cal-namespace="popupTeamLinkDarkTheme" data-cal-config='{"theme":"dark"}' data-cal-link="team/seeded-team/collective-seeded-team-event">Book with Test Team[Dark Theme]</button>
       <button data-cal-namespace="popupTeamLinksList" data-cal-link="team/seeded-team/">See Team Links [Auto Theme]</button>
+      <button data-cal-namespace="popupReschedule" data-cal-link="reschedule/qm3kwt3aTnVD7vmP9tiT2f">Reschedule Event[Auto Theme]</button>
```
Comment (hariombalhara) [inline packages/embeds/embed-core/index.html:91]: Add reschedule to testpage. Follow it up with a playwright test for reschedule.

File: packages/prisma/seed.ts [hunk 5]
```diff
@@ -242,6 +242,13 @@ async function main() {
             endTime: dayjs().add(2, "day").add(30, "minutes").toDate(),
             status: BookingStatus.PENDING,
           },
+          {
+            uid: "qm3kwt3aTnVD7vmP9tiT2f",
```
Comment (hariombalhara) [inline packages/prisma/seed.ts:247]: Seeded booking with fixed UUID to allow hardcoding the UID for rescheduling.

**Review by hariombalhara — COMMENTED** [review]
Self review done

**Comment by zomars** [comment]
Should we handle rescheduling from the new `/booking/[uid]` page as well? cc @CarinaWolli to hear her thoughts.

**Review by zomars — APPROVED** [review]
Thank you for your contribution @hariombalhara 🙏🏽

**Pull request merged**